src/Charcoal/Property/StorablePropertyTrait.php

```diff
@@ -121,6 +121,23 @@ public function fieldNames()
         return $this->fieldNames;
     }
 
+    /**
+     * Overridable to allow for custom Field Name parsing.
+     *
+     * This allows to add additional functions to a select statement that looks like this :
+     *  'MySQL_FUNCTION(select)'
+     *
+     * @param string $key       The property field key.
+     * @param mixed  $fieldName The raw filed name.
+     * @return mixed
+     */
+    protected function fieldExpression($key, $fieldName)
+    {
+        unset($key);
+
+        return $fieldName;
+    }
```

Comment:
This method needs some improvement in terms of input, output, and description.

For starters, I would improve the block comment of `fieldNames()` and `fieldValue()` to better hint what to expect from the methods:

```php
/**
 * Retrieve the property's namespaced storage field names.
 *
 * This method can be overridden for custom field names.
 *
 * Example #1 property `name`:
 *
 * ```
 * {
 *   "": "name_en"
 * }
 * ```
 *
 * Example #2 multilingual property `title`:
 *
 * ```
 * {
 *   "en": "title_en",
 *   "fr": "title_fr",
 *   "de": "title_de"
 * }
 * ```
 *
 * Example #3 custom property `obj`:
 *
 * ```
 * {
 *   "id": "obj_id",
 *   "type": "obj_type"
 * }
 * ```
 * 
 * Example #4 custom property `tuple`:
 *
 * ```
 * {
 *   "1": "tuple_1",
 *   "2": "tuple_2",
 *   "3": "tuple_3"
 * }
 * ```
 *
 * @return array<(string|int), string>
 */
public function fieldNames()
```

```php
/**
 * Format the property's value for the given field key suitable for storage.
 *
 * This method can be overridden for parsing custom field values.
 * It is recommended to override {@see self::storageVal()} instead.
 *
 * @param  string $key The property field key.
 * @param  mixed  $val The value to set as field value.
 * @return mixed
 */
protected function fieldValue($key, $val)
```

I propose the method name also explicitly state their SQL purpose.

```suggestion
    /**
     * Format the property's field name as an SQL SELECT expression.
     *
     * This method can be overridden for custom field name parsing.
     *
     * This method allows a property to apply an SQL function to a field
     * or alias a field:
     *
     * ```sql
     * ST_AsText(`geometry`)
     * ```
     *
     * ```sql
     * CONCAT(`first_name`, ' ', `last_name`) AS `full_name`
     * ```
     *
     * @param  string $key       The property field key.
     * @param  string $fieldName The field name.
     * @return string
     */
    protected function getSqlSelectExpression($key, $fieldName)
    {
        unset($key);

        return $fieldName;
    }
```